internal/envoy/auth.go

```diff
@@ -47,13 +47,18 @@ var (
 // UpstreamTLSContext creates an envoy_api_v2_auth.UpstreamTlsContext. By default
 // UpstreamTLSContext returns a HTTP/1.1 TLS enabled context. A list of
 // additional ALPN protocols can be provided.
-func UpstreamTLSContext(ca []byte, subjectName string, alpnProtocols ...string) *envoy_api_v2_auth.UpstreamTlsContext {
+func UpstreamTLSContext(ca []byte, subjectName string, sni string, alpnProtocols ...string) *envoy_api_v2_auth.UpstreamTlsContext {
 	context := &envoy_api_v2_auth.UpstreamTlsContext{
 		CommonTlsContext: &envoy_api_v2_auth.CommonTlsContext{
 			AlpnProtocols: alpnProtocols,
 		},
 	}
 
+	// If SNI is specified, set on the Upstream context
```

Comment:
Is this what we were asked to do? I though the request was "I want to change the host header"

Reply:
In my testing of the feature, if I proxy to an upstream that is secure TLS, then Envoy won't proxy and returns 503 to the caller. No, we weren't directly asked for this, but @jpeach and I found that setting this allows the proper proxying.

Reply:
Ok, thanks for explaining.

Reply:
👍